Lib/zipfile/__init__.py

```diff
@@ -603,6 +603,26 @@ def from_file(cls, filename, arcname=None, *, strict_timestamps=True):
 
         return zinfo
 
+    @classmethod
+    def for_name(cls, filename, archive, *, date_time=None):
```

Comment:
When I drafted this for zipfile.Path tests, I didn't put too much thought into how to name it or expose it publicly (if at all).

For example, should it be private (e.g. `_for_name`)? My instinct is yes, unless we have a use-case to expose it publicly. Then again, the need for it in zipfile.Path tests, suggests that maybe there is a public need.

Also, it's constructing a ZipInfo for a name and an archive. Maybe it should be named `for_name_and_archive` or maybe, since the archive is the more salient factor (not simply passed through to the regular constructor), it should be `for_archive`.

I don't feel strongly about it, but wanted to highlight these considerations.

Reply:
Actually, I struggled locally on how to name the method :') I went for `from_name`, `from_archive`, `for_name`, `for_name_and_archive`, `for_entry` but I eventually used what you already had :')

I'm also ok with not exposing it. It was a bit hard to come up with something clear in the docs and I'm not really happy with the current formulation.

Reply:
In the latest implementation, I'm moving forward with `_for_archive`. `archive` instead of `name` because it now is solely dependent on the `archive`. `for` instead of `from` because it's mutating an existing object and not constructing one. Private because let's not expose it more publicly until there's a demand for it.